packages/orchestration/src/types.d.ts

```diff
@@ -1,7 +1,470 @@
+ */
+export type CosmosChainInfo = {
+  chainId: string;
+  ibcConnectionInfo: {
+    id: string; // e.g. connection-0
+    client_id: string; // '07-tendermint-0'
+    state: 'OPEN' | 'TRYOPEN' | 'INIT' | 'CLOSED';
+    counterparty: {
+      client_id: string;
+      connection_id: string;
+      prefix: {
+        key_prefix: string;
+      };
+    };
+    versions: { identifier: string; features: string[] }[];
+    delay_period: bigint;
+  };
+  icaEnabled: boolean;
+  icqEnabled: boolean;
+  pfmEnabled: boolean;
+  ibcHooksEnabled: boolean;
+  /**
+   *
+   */
+  allowedMessages: TypeUrl[];
+  allowedQueries: TypeUrl[];
+};
+
+export type ChainInfo = CosmosChainInfo | EthChainInfo;
+
+// marker interface
+interface QueryResult {}
+
+/**
+ * An object for access the core functions of a remote chain.
+ *
+ * Note that "remote" can mean the local chain; it's just that
+ * accounts are treated as remote/arms length for consistency.
+ */
+export interface Chain<C extends keyof KnownChains> {
+  getChainInfo: () => Promise<KnownChains[C]['info']>;
+
+  // "makeAccount" suggests an operation within a vat
+  /**
+   * Creates a new account on the remote chain.
+   * @returns an object that controls a new remote account on Chain
+   */
+  createAccount: () => Promise<OrchestrationAccount<C>>;
+  // FUTURE supply optional port object; also fetch port object
+
+  /**
+   * Low level operation to query external chain state (e.g., governance params)
+   * @param queries
+   * @returns
+   *
+   */
+  query: (queries: Proto3JSONMsg[]) => Promise<Iterable<QueryResult>>;
+
+  /**
+   * Get the Denom on this Chain corresponding to the denom or Brand on
+   * this or another Chain.
+   * @param denom
+   * @returns
+   */
+  getLocalDenom: (denom: DenomArg) => Promise<Denom>;
+}
+
+/**
+ * Low level object that supports queries and operations for an account on a remote chain.
+ */
+export interface ChainAccount {
+  /**
+   * @returns the address of the account on the remote chain
+   */
+  getAccountAddress: () => string;
+  /**
+   * Submit a transaction on behalf of the remote account for execution on the remote chain.
+   * @param msgs - records for the transaction
+   * @returns acknowledgement string
+   */
+  executeTx: (msgs: Proto3JSONMsg[]) => Promise<string>;
+  /**
+   * Submit a transaction on behalf of the remote account for execution on the remote chain.
+   * @param msgs - records for the transaction
+   * @returns acknowledge string
+   */
+  executeEncodedTx: (msgs: AnyJson[]) => Promise<string>;
+  /** deposit payment from zoe to the account*/
+  deposit: (payment: Payment) => Promise<void>;
+  /** get Purse for a brand to .withdraw() a Payment from the account */
+  getPurse: (brand: Brand) => Promise<Purse>;
+  /**
+   * Close the remote account
+   */
+  close: () => Promise<void>;
+  /* transfer account to new holder */
+  prepareTransfer: () => Promise<Invitation>;
+}
+
+export interface Undelegation {
+  cancel: () => Promise<void>;
```

Comment:
```suggestion 
  cancel: () => Promise<MsgCancelUnbondingDelegationResponse>;
```

Would also need to add this type to `delegation.d.ts`:
```ts
type MsgCancelUnbondingDelegationResponse {}; 
```

_This type/message does not appear in our current `@agoric/cosmic-proto`, and was seemingly [released](https://github.com/cosmos/cosmos-sdk/blob/30383e6cf4f584d7e9fc4cf79508955c9a25d1f2/x/staking/proto/cosmos/staking/v1beta1/tx.proto#L190) in cosmos-sdk v0.46.x_